lib/BridgedRoom.js

```diff
@@ -412,10 +412,11 @@ BridgedRoom.prototype.onMatrixMessage = async function(message) {
     sendMessageParams.body.username = user.getDisplaynameForRoom(message.room_id);
 
     const reply = await this.findParentReply(message);
+    let parentEvent = undefined;
     if (reply !== message.event_id) {
         // We have a reply
-        const parentStoredEvent = await store.getEntryByMatrixId(message.room_id, reply);
-        sendMessageParams.body.thread_ts = parentStoredEvent.remoteEventId;
+        parentEvent = await store.getEntryByMatrixId(message.room_id, reply);
+        sendMessageParams.body.thread_ts = parentEvent.remoteEventId;
```

Comment:
What happens if `parentEvent` is undefined?

Reply:
Then something has gone horribly wrong and we don't have the event that's being replied to?

Reply:
We should definitely give up at that point and throw an error then, we can't assume we have it.